### PR TITLE
Feat/client/parallax background

### PR DESCRIPTION
**Pull request by EmileEttel**
Inclusion of parallax component and animated component which allows sprites to be animated

### CODE REVIEW

File: src/client/entrypoint.cpp [hunk 1]
```diff
@@ -55,6 +61,57 @@ ecs::World getGameWorld(ecs::Engine &engine)
...
+    world.addSystem(ecs::systems::HandleParallaxBounds);
+
+    ecs::Entity background1 = world.registry.spawn_entity();
+    ecs::Entity backgroundp1 = world.registry.spawn_entity();
+    ecs::Entity background2 = world.registry.spawn_entity();
+    ecs::Entity backgroundp2 = world.registry.spawn_entity();
+    ecs::Entity background3 = world.registry.spawn_entity();
+    ecs::Entity backgroundp3 = world.registry.spawn_entity();
+
+    world.registry.addComponent<ecs::component::Position>(background1, {0, 0});
```
Comment (GuyomT) [inline src/client/entrypoint.cpp:73]: @aurelenc will confirm but their is a new way to load asset, anyway its working atm need to merge AssetLoader

**Review by GuyomT — APPROVED** [review]
LGTM

File: src/client/entrypoint.cpp (hunk 1 above)
Comment (EmileEttel) [reply src/client/entrypoint.cpp:73]: there*

File: src/ecs/components/client/Animated.hpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+     * Animated component allows draw system function to render the animation for the sprite
+     */
+    struct Animated {
+        int width;
+        int origin_x;
+        int origin_y;
+        int current;
+        int freq;
+        int cur_freq;
+    };
```
Comment (Cavonstavant) [inline src/ecs/components/client/Animated.hpp:22]: Needs more doc on `Animated` struct members. We can't really guess their uses just by their name

File: src/ecs/components/client/Parallax.hpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+#pragma once
+
+namespace ecs::component
+{
+    /**
+     * Parallax component allows us to more easily manipulate the parallaxed background
+     */
+    struct Parallax {
+        int threshold;
+        int position;
```
Comment (Cavonstavant) [inline src/ecs/components/client/Parallax.hpp:17]: same here

File: src/ecs/systems/client/Draw.hpp [hunk 4]
```diff
@@ -28,17 +29,35 @@ namespace ecs::systems
...
+                        anim.value().cur_freq = 0;
+                        anim.value().current++;
+                        if (anim.value().current * anim.value().width + anim.value().origin_x >= draw.value().Texture.getSize().x)
+                            anim.value().current = 0;
+                    }
+                    scaleX = static_cast<float>(size.value().width) / static_cast<float>(anim.value().width);
+                    scaleY = static_cast<float>(size.value().height) / static_cast<float>(draw.value().Texture.getSize().y);
+                } else {
+                    scaleX = static_cast<float>(size.value().width) / static_cast<float>(draw.value().Texture.getSize().x);
+                    scaleY = static_cast<float>(size.value().height) / static_cast<float>(draw.value().Texture.getSize().y);
```
Comment (Cavonstavant) [inline src/ecs/systems/client/Draw.hpp:59]: potential div. by zero.
```suggestion
                    if (static_cast<float>(draw.value().Texture.getSize().x != 0)
                        scaleX = static_cast<float>(size.value().width) / static_cast<float>(draw.value().Texture.getSize().x);
                    else
                        scaleX = 0;
                    if (static_cast<float>(draw.value().Texture.getSize().y != 0)
                        scaleY = static_cast<float>(size.value().height) / static_cast<float>(draw.value().Texture.getSize().y);
                    else
                        scaleY = 0;
```
or smth like that

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
Need more docs and safety checks

**Pull request merged**